### DISCUSSION TITLE
"Unknown Log Format Character '<' Runtime Error in OMNeT++ Veins Simulation"

### DISCUSSION THREAD

**Discussion by 12shuaibi** [omnetpp/omnetpp#1148] [General]
I'm experiencing a runtime error when trying to run a Veins simulation using OMNeT++ 5.1.1, SUMO 0.30.0, and Veins 4.6. The error message is:
'Starting...

$ cd D:/home/omnetpp-5.1.1/samples/veins/examples/veins
$ ../../../../bin/opp_run.exe -m -n .;../../src/veins --image-path=../../images -l ../../src/veins omnetpp.ini

OMNeT++ Discrete Event Simulation  (C) 1992-2017 Andras Varga, OpenSim Ltd.
Version: 5.1.1, build: 170508-adbabd0, edition: Academic Public License -- NOT FOR COMMERCIAL USE
See the license for distribution terms and warranty disclaimer

Setting up Qtenv...

Loading NED files from .:  1
Loading NED files from ../../src/veins:  33
Loading NED files from D:\home\omnetpp-5.1.1\samples\veins\examples\veins:  1

Loading images from 'D:\home\omnetpp-5.1.1\samples\veins\images': *: 0  veins/*: 0  veins/node/*: 40  veins/sign/*: 5 
Loading images from 'D:\home\omnetpp-5.1.1\images': *: 0  abstract/*: 90  background/*: 4  block/*: 320  device/*: 195  logo/*: 1  maps/*: 9  misc/*: 70  msg/*: 55  old/*: 111  status/*: 28 

RUNTIME ERROR. A cRuntimeError exception is about to be thrown, and you
requested (by setting debug-on-errors=true in the ini file) that errors
abort execution and break into the debugger.

You should now probably be running the simulation under gdb or another
debugger. The simulation kernel will now raise a SIGINT signal which will
get you into the debugger. If you are not running under a debugger, you can
still use the core dump for post-mortem debugging. Once in the debugger,
view the call stack (in gdb: "bt" command) to see the context of the
runtime error.

<!> Unknown log format character '<'

TRAPPING on the exception above, due to a debug-on-errors=true configuration option. Is your debugger ready?

Simulation terminated with exit code: -2147483645
Working directory: D:/home/omnetpp-5.1.1/samples/veins/examples/veins
Command line: ../../../../bin/opp_run.exe -m -n .;../../src/veins --image-path=../../images -l ../../src/veins omnetpp.ini

Environment variables:
PATH=;D:/home/omnetpp-5.1.1/samples/veins/src;D:\home\omnetpp-5.1.1\bin;D:\home\omnetpp-5.1.1\tools\win64\mingw64\bin;D:\home\omnetpp-5.1.1\tools\win64\usr\bin;;D:/home/omnetpp-5.1.1/ide/jre/bin/server;D:/home/omnetpp-5.1.1/ide/jre/bin;D:/home/omnetpp-5.1.1/ide/jre/lib/amd64;.;D:\home\omnetpp-5.1.1\bin;D:\home\omnetpp-5.1.1\tools\win64\mingw64\bin;D:\home\omnetpp-5.1.1\tools\win64\usr\local\bin;D:\home\omnetpp-5.1.1\tools\win64\usr\bin;D:\home\omnetpp-5.1.1\tools\win64\usr\bin;C:\Windows\System32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;D:\home\omnetpp-5.1.1\tools\win64\usr\bin\site_perl;D:\home\omnetpp-5.1.1\tools\win64\usr\bin\vendor_perl;D:\home\omnetpp-5.1.1\tools\win64\usr\bin\core_perl;D:\home\veins-veins-4.6\examples\veins;
OMNETPP_ROOT=D:/home/omnetpp-5.1.1/
OMNETPP_IMAGE_PATH=D:\home\omnetpp-5.1.1\images'

I've tried checking the omnetpp.ini configuration file and the model files for syntax errors, but I haven't been able to identify the issue. I've also tried cleaning and recompiling the project, but the problem persists."

**Comment by avarga** [comment]
This comes from a .qtenvrc file saved by another version of OMNeT++. Just
delete it and it'll be fine.

Andras

**Reply by 12shuaibi** [reply]
Okay, thank you. I have deleted the .qtenvrc file in the D:\home\omnetpp-5.1.1\samples\aloha path, and I haven't found any other .qtenvrc files. I ran the omnetpp.ini file, but I still got the same error, as well as the error message shown in the following image.
<img width="269" alt="屏幕截图 2024-04-11 192820" src="https://github.com/omnetpp/omnetpp/assets/150670093/ed7f8f86-ad6e-4bc9-8660-a026c568bde8">

**Reply by torokati44** [reply]
There is another, global `.qtenvrc` file in the location documented here: https://doc.qt.io/qt-6/qdir.html#homePath
This setting is stored there.

**Reply by 12shuaibi** [reply]
Thank you very much, I have been able to run it successfully.
